src/contract/aci.js

```diff
@@ -235,7 +235,7 @@ export default async function getContractInstance ({
     if (!fn) throw new MissingFunctionNameError()
     if (fn === 'init' && !opt.callStatic) throw new InvalidMethodInvocationError('"init" can be called only via dryRun')
     if (!contractId && fn !== 'init') throw new InvalidMethodInvocationError('You need to deploy contract before calling!')
-    if (opt.amount > 0 && fnACI.payable === false) throw new NotPayableFunctionError(`You try to pay "${opt.amount}" to function "${fn}" which is not payable. Only payable function can accept coins`)
+    if (fn !== 'init' && opt.amount > 0 && fnACI.payable === false) throw new NotPayableFunctionError(`You try to pay "${opt.amount}" to function "${fn}" which is not payable. Only payable function can accept coins`)
```

Comment:
would work to change `{ payable: false }` to `{ payable: true }` in `getFunctionACI` instead?

Reply:
Function call sometimes never toches that line when the function ACI is already present in the instance. if statement in here.  https://github.com/aeternity/aepp-sdk-js/blob/5fe6cac41718fd7dd8645a1cf5034bdb654067a6/src/contract/aci.js#L215

Reply:
we can do it later after implementing https://github.com/aeternity/aesophia/issues/373

Reply:
Do we need a new issue here regarding `init` is  `payble` or not?

Reply:
from current discussion in that issue above  `init` will never be `payable`. we should discuss that in the issue @davidyuk opened in aesophia repo.

I guess current solution here is good as is probably. need your statements about that

Reply:
Yeah, I read the thread in aesophia issue. `init` is never callable which means we can't estimate gas for deployment as of now which is wrong, we should be able to estimate gas for deployment calls. 

My conclusion is: Let's keep the changes. Ideally, the compiler throws an error anyway. the source of truth is with the compiler/node. That means the following client-side check actually doesn't make sense. It's just saving a request from the compiler. The only problem I see in here with the fix we did is **SDK now allows sending amount during deployment call even if the contract is not marked as payable**. Well, the node will return an error anyway😀
https://github.com/aeternity/aepp-sdk-js/blob/5fe6cac41718fd7dd8645a1cf5034bdb654067a6/src/contract/aci.js#L238

Reply:
> we can't estimate gas for deployment as of now which is wrong, we should be able to estimate gas for deployment calls

who said that? actually it is possible to estimate gas afaik.

> The only problem I see in here with the fix we did is **SDK now allows sending amount during deployment call even if the contract is not marked as payable**. Well, the node will return an error anyway😀

I agree, but if we don't do it people can't provide AE even to `payable` contracts, right? ... oh man 😵 

> Well, the node will return an error anyway😀

it won't !!!! this is the reason why @davidyuk opened the issue in aesophia repository I guess. we actually can send AE to a non-payable contract during deployment with the `ContractCreateTx`.

mhhh 🤔 

what do you think, @kenodressel @thepiwo ?

Reply:
> who said that? actually it is possible to estimate gas afaik.  

I mean without the fix in this PR. 

> I agree, but if we don't do it people can't provide AE even to payable contracts, right? ... oh man. 

Yes. They still can provided `gas` option.

> it won't !!!! this is the reason why @davidyuk opened the issue in aesophia repository I guess. we actually can send AE to a non-payable contract during deployment with the ContractCreateTx

Wow! I hope this is the case only for `contractCreateTx`. Imagine compiler depending on client call ACI for a contract method. This unblocks a new fear of attack. A contract method ACI can be marked payable at the client-side and so on..

Reply:
> Wow! I hope this is the case only for `contractCreateTx`. Imagine compiler depending on client call ACI for a contract method. This unblocks a new fear of attack. A contract method ACI can be marked payable at the client-side and so on..

yes, only for deployment (or `init`, but we shouldn't say `init` is used because it somehow has no meaning except setting the initial state)

Reply:
The current version is fine 👌